### PR TITLE
dird: fix tls psk connection failure with long job names

### PR DESCRIPTION
**Pull request by alaaeddineelamri**
#### Description
When having very long job names exceeding 98 characters, we start seeing a TLS-PSK failure with error 1115. This is due to a mix of psk identity limitations set by OpenSSL and truncation code we have.
This PR fixes the issue.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted
- ~~[ ] If backport: add original PR number and target branch at top of this file: **Backport of PR#000 to bareos-2x**~~

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault
- [x] Manually tested dir/sd openSSL3x with fd openSSL1x backup restore status
- [x] Manually tested dir/sd openSSL1x with fd openSSL3x backup restore status
- [x] Manually tested dir openSSL3x / sd openSSL1x / fd openSSL1x
- [x] Manually tested dir openSSL1x / sd openSSL3x / fd openSSL1x

### CODE REVIEW

File: docs/manuals/source/manually_added_config_directive_descriptions/dir-job-Name.rst.inc [hunk 1]
```diff
@@ -1,3 +1,7 @@
 The Job name. This name can be specified on the Run command in the console program to start a job. If the name contains spaces, it must be specified between quotes. It is generally a good idea to give your job the same name as the Client that it will backup. This permits easy identification of jobs.
 
-When the job actually runs, the unique Job Name will consist of the name you specify here followed by the date and time the job was scheduled for execution. This directive is required.
+When the job actually runs, the unique Job Name will consist of the name you specify here followed by the date and time the job was scheduled for execution.
+
+It is recommended to limit job names to 98 characters. Higher is always possible, but when the job is run, its name will be truncated to accomodate certain protocol limitations, as well as the above mentioned date and time.
+
+This directive is required.
```
Comment (joergsteffens) [inline docs/manuals/source/manually_added_config_directive_descriptions/dir-job-Name.rst.inc:7]:
```suggestion
```
That the directive is required will be written automatically to the documentation.

**Review by bruno-at-bareos — APPROVED** [review]
Thanks really good job, I like the approach which is also quite easy to backport if needed.
To be merge once green.

**Comment by bruno-at-bareos** [comment]
Reminder: test cross openssl before merge

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**